scalameter-core/src/main/scala/org/scalameter/Context.scala

```diff
@@ -20,6 +20,9 @@ case class Context(properties: immutable.Map[Key[_], Any]) {
     }
   }
   def goe[T](key: Key[T], v: => T) = properties.getOrElse(key, v).asInstanceOf[T]
+  def apply[T](key: KeyWithDefaultKey[T]) = {
```

Comment:
You could use intersection types, or you could introduce their common superclass, something like `UbiquitousKey` -- I'm taking suggestions for the name.